### PR TITLE
more supported and unsupported features

### PR DESCRIPTION
**Pull request by y1ca1**
Doc more trait features and associated constants

### CODE REVIEW

**Review by parno — APPROVED** [review]
Looks reasonable to me

File: source/docs/guide/src/features.md [hunk 1]
```diff
@@ -312,6 +316,14 @@ Note that Verus is in active development. If a feature is unsupported, it might
     <td>User-defined traits</td>
     <td>Supported</td>
   </tr>
+  <tr>
+    <td>Default implementations</td>
+    <td>Supported</td>
+  </tr>
+  <tr>
+    <td>Super-traits</td>
```
Comment (utaal) [inline source/docs/guide/src/features.md:324]:
```suggestion
    <td>Trait bounds on trait declarations</td>
```

File: source/docs/guide/src/features.md [hunk 2]
```diff
@@ -320,6 +332,14 @@ Note that Verus is in active development. If a feature is unsupported, it might
     <td>Associated types</td>
     <td>Partially supported</td>
   </tr>
+  <tr>
+    <td>Generic associated types</td>
+    <td>Partially supported (with lifetimes)</td>
```
Comment (utaal) [inline source/docs/guide/src/features.md:337]:
```suggestion
    <td>Partially supported (only lifetimes are supported)</td>
```

**Review by utaal — APPROVED** [review]

**Comment by utaal** [comment]
Can you please review and apply the suggested changes? thanks!

**Pull request merged**